diff --git a/drivers/video/v4l2driver.cpp b/drivers/video/v4l2driver.cpp
index cf5fe6b50..e670c6ffd 100644
--- a/drivers/video/v4l2driver.cpp
+++ b/drivers/video/v4l2driver.cpp
@@ -946,6 +946,14 @@ bool V4L2_Driver::StartExposure(float duration)
 
 bool V4L2_Driver::setShutter(double duration)
 {
+    exposure_duration.tv_sec  = (long) duration;
+    exposure_duration.tv_usec = (long) ((duration - (double) exposure_duration.tv_sec) * 1000000.0f);
+
+    elapsed_exposure.tv_sec = 0;
+    elapsed_exposure.tv_usec = 0;
+
+    gettimeofday(&capture_start, nullptr);
+
     if (lx->isEnabled())
     {
         LOGF_INFO("Using long exposure mode for %.3f sec frame.", duration);
@@ -963,14 +971,6 @@ bool V4L2_Driver::setShutter(double duration)
     }
     else if (setManualExposure(duration))
     {
-        exposure_duration.tv_sec  = (long) duration;
-        exposure_duration.tv_usec = (long) ((duration - (double) exposure_duration.tv_sec) * 1000000.0f);
-
-        elapsed_exposure.tv_sec = 0;
-        elapsed_exposure.tv_usec = 0;
-
-        gettimeofday(&capture_start, nullptr);
-
         frameCount    = 0;
         subframeCount = 0;
 
@@ -992,6 +992,11 @@ bool V4L2_Driver::isIOptron()
     return strstr(getDeviceName(), "iGuider") || strstr(getDeviceName(), "iPolar");
 }
 
+bool V4L2_Driver::isSpc900Nc()
+{
+    return strstr(v4l_base->getDeviceName(), "SPC 900NC");
+}
+
 bool V4L2_Driver::setManualExposure(double duration)
 {
     /* N.B. Check how this differs from one camera to another. This is just a proof of concept for now */
@@ -1537,7 +1542,7 @@ void V4L2_Driver::newFrame()
         return;
     }
 
-    if ( PrimaryCCD.isExposing() )
+    if ( PrimaryCCD.isExposing() || isSpc900Nc() )
     {
         non_capture_frames = 0;
         if( !is_capturing )
diff --git a/drivers/video/v4l2driver.h b/drivers/video/v4l2driver.h
index a1f010922..1c260b214 100644
--- a/drivers/video/v4l2driver.h
+++ b/drivers/video/v4l2driver.h
@@ -198,6 +198,8 @@ public:
 
         void resetDevice(int bus_num, int dev_num);
 
+        bool isSpc900Nc();
+
         /* Variables */
         INDI::V4L2_Base *v4l_base;
 
