diff --git a/drivers/video/v4l2driver.cpp b/drivers/video/v4l2driver.cpp
index cf5fe6b50..08901312c 100644
--- a/drivers/video/v4l2driver.cpp
+++ b/drivers/video/v4l2driver.cpp
@@ -916,8 +916,7 @@ bool V4L2_Driver::StartExposure(float duration)
         V4LFrame->expose = duration;
         PrimaryCCD.setExposureDuration(duration);
 
-        if (!lx->isEnabled() || lx->getLxmode() == LXSERIAL)
-            start_capturing(false);
+        start_capturing(false);
 
         /* Update exposure duration in client */
         /* FIXME: exposure update timer has period hardcoded 1 second */
@@ -951,8 +950,9 @@ bool V4L2_Driver::setShutter(double duration)
         LOGF_INFO("Using long exposure mode for %.3f sec frame.", duration);
         if (startlongexposure(duration))
         {
+            frame_duration.tv_sec  = (long) duration;
+            frame_duration.tv_usec = (long) ((duration - (double) ((long) duration)) * 1000000.0f);
             LOGF_INFO("Started %.3f-second long exposure.", duration);
-            return true;
         }
         else
         {
@@ -963,21 +963,9 @@ bool V4L2_Driver::setShutter(double duration)
     }
     else if (setManualExposure(duration))
     {
-        exposure_duration.tv_sec  = (long) duration;
-        exposure_duration.tv_usec = (long) ((duration - (double) exposure_duration.tv_sec) * 1000000.0f);
-
-        elapsed_exposure.tv_sec = 0;
-        elapsed_exposure.tv_usec = 0;
-
-        gettimeofday(&capture_start, nullptr);
-
-        frameCount    = 0;
-        subframeCount = 0;
-
-        // Do not spam log for short exposures.
+        // frame_duration already set by setManualExposure()
         if (duration >= 3)
             LOGF_INFO("Started %.3f-second manual exposure.", duration);
-        return true;
     }
     else
     {
@@ -985,6 +973,15 @@ bool V4L2_Driver::setShutter(double duration)
                   duration);
         return false;
     }
+
+    exposure_duration.tv_sec  = (long) duration;
+    exposure_duration.tv_usec = (long) ((duration - (double) exposure_duration.tv_sec) * 1000000.0f);
+    elapsed_exposure.tv_sec   = 0;
+    elapsed_exposure.tv_usec  = 0;
+    gettimeofday(&capture_start, nullptr);
+    frameCount    = 0;
+    subframeCount = 0;
+    return true;
 }
 
 bool V4L2_Driver::isIOptron()
@@ -1311,14 +1308,7 @@ void V4L2_Driver::lxtimerCallback(void * userpointer)
     auto p = static_cast<V4L2_Driver *>(userpointer);
 
     p->lx->stopLx();
-    if (p->lx->getLxmode() == LXSERIAL)
-    {
-        p->v4l_base->setlxstate(LX_TRIGGERED);
-    }
-    else
-    {
-        p->v4l_base->setlxstate(LX_ACTIVE);
-    }
+    p->v4l_base->setlxstate(LX_TRIGGERED);
     IERmTimer(p->lxtimer);
     if (!p->v4l_base->isstreamactive())
         p->is_capturing = p->start_capturing(false); // jump to new/updateFrame
@@ -1800,7 +1790,15 @@ bool V4L2_Driver::AbortExposure()
 {
     if (lx->isEnabled())
     {
+        if (lxtimer != -1)
+        {
+            IERmTimer(lxtimer);
+            lxtimer = -1;
+        }
         lx->stopLx();
+        v4l_base->setlxstate(LX_ACTIVE);
+        if (!Streamer->isBusy())
+            stop_capturing();
         return true;
     }
     else if (!Streamer->isBusy())
