diff --git a/drivers/video/v4l2driver.cpp b/drivers/video/v4l2driver.cpp
index cf5fe6b50..b1b96f820 100644
--- a/drivers/video/v4l2driver.cpp
+++ b/drivers/video/v4l2driver.cpp
@@ -951,6 +951,7 @@ bool V4L2_Driver::setShutter(double duration)
         LOGF_INFO("Using long exposure mode for %.3f sec frame.", duration);
         if (startlongexposure(duration))
         {
+            gettimeofday(&capture_start, nullptr);
             LOGF_INFO("Started %.3f-second long exposure.", duration);
             return true;
         }
@@ -992,6 +993,11 @@ bool V4L2_Driver::isIOptron()
     return strstr(getDeviceName(), "iGuider") || strstr(getDeviceName(), "iPolar");
 }
 
+bool V4L2_Driver::isSpc900Nc()
+{
+    return strstr(v4l_base->getDeviceName(), "SPC 900NC");
+}
+
 bool V4L2_Driver::setManualExposure(double duration)
 {
     /* N.B. Check how this differs from one camera to another. This is just a proof of concept for now */
@@ -1537,7 +1543,7 @@ void V4L2_Driver::newFrame()
         return;
     }
 
-    if ( PrimaryCCD.isExposing() )
+    if ( PrimaryCCD.isExposing() || isSpc900Nc() )
     {
         non_capture_frames = 0;
         if( !is_capturing )
diff --git a/drivers/video/v4l2driver.h b/drivers/video/v4l2driver.h
index a1f010922..1c260b214 100644
--- a/drivers/video/v4l2driver.h
+++ b/drivers/video/v4l2driver.h
@@ -198,6 +198,8 @@ public:
 
         void resetDevice(int bus_num, int dev_num);
 
+        bool isSpc900Nc();
+
         /* Variables */
         INDI::V4L2_Base *v4l_base;
 
